flow/record/adapter/avro.py

```diff
@@ -121,8 +123,12 @@ def __iter__(self) -> Iterator[record.Record]:
                     obj[field_name] = EPOCH + timedelta(microseconds=value)
 
             rec = self.desc.recordType(**obj)
+            ctx.records_read += 1
             if not self.selector or self.selector.match(rec):
```

Comment:
Maybe an idea that you can "attach" the context to the selector, and let the selector increment `records_matched` and `records_excluded`? Easiest way would be to just add a second argument.

Reply:
I opted for a helper function called `match_record_with_context` in the selector namespace.

One could argue it can also increase `ctx.read`, then you don't have to do any metrics tracking in the adapter itself besides just calling this function on every record.

Reply:
> One could argue it can also increase `ctx.read`, then you don't have to do any metrics tracking in the adapter itself besides just calling this function on every record.

Hm if you do that, I'd maybe move the helper function to `flow.record.context` instead.